### DISCUSSION TITLE
Cannot get custom keywords to work

### DISCUSSION THREAD

**Discussion by BWindey** [folke/todo-comments.nvim#329] [Q&A] [answered]
Hi, I really like the highlighting, but I cannot get custom keywords to work for my config.

I'm using Lazy, and the plugin is on the latest version.
My neovim is on v0.10.2

This is my config-file:
```lua
return {
	{
		"folke/todo-comments.nvim",
		dependencies = { "nvim-lua/plenary.nvim" },
		opts = {
			keywords = {
				ASK = { icon = "?", color = "hint", alt = { "QUESTION" } },
			},
			merge_keywords = true,
		},
		-- TODO:
		--
		-- ASK: th

		config = function ()
			local wk = require("which-key")

			wk.add({
			mode = 'n',
				{ "<leader>t", group = "Todo's..." },
				{ "<leader>td", "<CMD>TodoQuickFix<CR>", desc = "QuickFix-list with all Todo's" },
				{ "<leader>tt", "<CMD>TodoTelescope<CR>", desc = "Telescope with all Todo's" },
			})

			require("todo-comments").setup()
		end
	}
}
```
And here you can see that the custom keyword does not work:
![image](https://github.com/user-attachments/assets/6a4149a8-81cd-40a2-b066-c9fbb4679da4)

Is there something I'm doing wrong, or do I have bad luck with outside factors?

Setting "ASK" as an alt for "NOTE" also does not work.

**Comment by BWindey** [comment, marked as answer]
Welp, just answered my own question.
I think it could have been better explained in the docs, but you don't need to edit the `ops` table, but you have to pass the custom configuration to the `setup()` function inside `config = { ... }`.

My update config:
```lua
return {
	{
		"folke/todo-comments.nvim",
		dependencies = { "nvim-lua/plenary.nvim" },
		-- TODO:
		--
		-- ASK: th

		config = function ()
			local wk = require("which-key")

			wk.add({
			mode = 'n',
				{ "<leader>t", group = "Todo's..." },
				{ "<leader>td", "<CMD>TodoQuickFix<CR>", desc = "QuickFix-list with all Todo's" },
				{ "<leader>tt", "<CMD>TodoTelescope<CR>", desc = "Telescope with all Todo's" },
			})

			require("todo-comments").setup({
				keywords = {
					ASK = { icon = "?", color = "hint", alt = { "QUESTION" } },
				},
			})
		end
	}
}
```

And now they render:
![image](https://github.com/user-attachments/assets/1824a4bf-7bc9-4072-8e07-6b7ba03e85cd)

**Comment by Kaya-Sem** [comment]
I had the same issue as @BWindey, but it's fixed now. +1 for the docs
